### ISSUE TITLE
Neorg not highlighting markup

### ISSUE THREAD

**Issue by MihailMihov** [nvim-neorg/neorg#1388]
### Prerequisites

- [X] I am using the latest stable release of Neovim
- [X] I am using the latest version of the plugin

### Neovim Version

NVIM v0.10.0-dev-2783+g381806729-dirty Build type: Release LuaJIT 2.1.0-beta3

### Neorg setup

```lua
require("neorg").setup({
	load = {
		["core.defaults"] = {},
		["core.concealer"] = {},
		["core.dirman"] = {
			config = {
				workspaces = {
					dev = "~/dev/notes",
				},
				index = "index.norg",
			}
		}
	}
})
```

### Actual behavior

![image](https://github.com/nvim-neorg/neorg/assets/25751178/d30d6375-a79e-4204-92e1-ca96bf37b891)
![image](https://github.com/nvim-neorg/neorg/assets/25751178/c4cacb08-c104-4536-a52a-a556fa136c4f)
First screenshot is before `:Neorg toggle-concealer`, second is after. Only the headings change to icons and for some reason the spoiler text. Same issue as #1374.

### Expected behavior

Markdown text is properly highlighted.

### Steps to reproduce

I followed the instructions to install `rocks.nvim`. My config is [here](https://github.com/MihailMihov/dotfiles/tree/main/.config/nvim). I set my colorscheme as opt and loaded it in it's rocks-config file, because otherwise I got an error that it's not found (as per the rocks-config [README](https://github.com/nvim-neorocks/rocks-config.nvim?tab=readme-ov-file#initialization-order)). I was then experiencing the issue in #1351, so I added the snipper from there to my config and now neorg loads, but the concealer isn't working properly. Running `:InspectTree` does give the expected output I think:
![image](https://github.com/nvim-neorg/neorg/assets/25751178/52b158f5-73a5-442e-ace8-fe064c4866f2)

### Other information

I'm using foot terminal, I have Source Code Pro and the symbols from Nerd Fonts installed. I also installed the tree-sitter-cli, but that didn't change anything. I tried Vhyrro's config, as it uses rocks.nvim, but I believe that it isn't updated as it failed to even load the colorscheme.

### Help

No

**Comment by flexagoon** [comment]
It seems like you don't have TreeSitter highlighting enabled

https://github.com/MihailMihov/dotfiles/blob/main/.config/nvim/lua/plugins/nvim-treesitter.lua

Try adding

```lua
highlight = { enable = true }
```

To nvim-treesitter opts

**Comment by MihailMihov** [comment]
Thank you so much that was it. Once the issues with `rocks-treesitter.nvim` are resolved and it can be used instead of `nvim-treesitter` will this option be enabled automatically? Also could I add a healthcheck to the core.concealer module to warn users if they don't have treesitter highlighting enabled?

**Comment by flexagoon** [comment]
@MihailMihov I'm not a maintainer of neorg or rocks-treesitter, so I don't know for sure, but looking at the rocks-treesitter readme, it seems like highlighting should still be enabled in the `rocks.toml` file:

https://github.com/nvim-neorocks/rocks-treesitter.nvim?tab=readme-ov-file#using-rockstoml

**Comment by vhyrro** [comment]
@MihailMihov yep once rocks-treesitter.nvim is fully stabilized highlighting will be enabled in all files automatically. In the meantime you can set your colourscheme from rocks-config.nvim itself:

```toml
[config]
colorscheme = "kanagawa-wave" # Example colourscheme
```

This is equivalent to running `vim.cmd.colorscheme("kanagawa-wave")` but rocks-config.nvim makes sure to run this at a time when the colourscheme is available :)
